### PR TITLE
Fix the error caused by passing unused context in JiraSensor.poke

### PR DESCRIPTION
**Pull request by kianelbo**
---
#### This tiny PR attempts to fix the error caused by https://github.com/apache/airflow/pull/23046
(original issue: #22810 )

### CODE REVIEW

File: airflow/providers/jira/sensors/jira.py [hunk 1]
```diff
@@ -60,7 +60,7 @@ def poke(self, context: 'Context') -> Any:
         jira_result = getattr(resource, self.method_name)(**self.method_params)
         if self.result_processor is None:
             return jira_result
-        return self.result_processor(context, jira_result)
```
Comment (jhtimmins) [inline airflow/providers/jira/sensors/jira.py:63]: Is there a reason we're not removing `context` from the `poke` method signature?

Comment (uranusjr) [reply airflow/providers/jira/sensors/jira.py:63]: `poke` on other sensors take `context`, so this probably should as well.

**Comment by ashb** [comment]
Could you add a test case to prevent future regressions?

**Comment by kianelbo** [comment]
> Could you add a test case to prevent future regressions?

Good idea! However I couldn't find a neat way to add a new case to the test. Instead, I thought it would be worthwhile to use the original `field_checker_func` of `JiraTicketSensor` and have a more realistically mocked ticket object.

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
Rebased to rebuild. @kianelbo - please ping us if you see it's green or when you fix it, if it is red :).

**Comment by potiuk** [comment]
Thanks @kianelbo !

**Pull request merged**